ChatItirod/ChatItirod/Program.cs

```diff
@@ -75,8 +77,33 @@ private static void ReceiveMessage()
                 {
                     byte[] data = receiver.Receive(ref remoteIp); // получаем данные
                     string message = Encoding.Unicode.GetString(data);
+                    if (message.Contains(":"))
+                    {
+                        Chat();
+                    }
+                    else
+                    {
+                        int index = Int32.Parse(message);
+                        var resendMessages = order.Where(x => x.Key >= index);
```

Comment:
How can you be so sure that all those messages were lost? You have an index of only one lost packet.